test/zimwriterfs-zimcreatorfs.cpp

```diff
@@ -127,3 +128,49 @@ TEST(ZimCreatorFSTest, ThrowsErrorIfDirectoryNotExist)
     ZimCreatorFS zimCreator("Non-existing-dir");
   }, std::invalid_argument );
 }
+
+struct Redirect {
+  std::string path;
+  std::string title;
+  std::string target;
+};
+
+bool operator==(const Redirect& a, const Redirect& b) {
+  return a.path == b.path && a.title == b.title && a.target == b.target;
+}
+
+void parse_redirectArticles(std::istream& in_stream, std::function<void(std::string, std::string, std::string)> handler);
+
+TEST(ZimCreatorFSTest, ParseRedirect)
+{
+  {
+  std::stringstream ss;
+  ss << "path\ttitle\ttarget\n";
+  ss << "A/path/to/somewhere\tAn amazing title\tAnother/path";
+
+  std::vector<Redirect> found;
+  parse_redirectArticles(
+    ss,
+    [&](std::string path, std::string title, std::string target)
+     {found.push_back({path, title, target});}
+  );
+
+  std::vector<Redirect> expected;
+  expected.push_back({"path", "title", "target"});
+  expected.push_back({"A/path/to/somewhere", "An amazing title", "Another/path"});
+  EXPECT_EQ(found, expected);
+  }
+
+
+  {
+    std::stringstream ss;
+    ss << "A/path\tOups, no target";
+    EXPECT_THROW({
+      parse_redirectArticles(
+            ss,
+            [&](std::string path, std::string title, std::string target)
+             {}
+          );
+    }, std::runtime_error);
+  }
```

Comment:
Please also add a test with more than 3 fields (more than 2 tab separators).

Reply:
Done.
And, there was a bug here. Now fixed.